go/common/grpc/proxy/proxy.go

```diff
@@ -66,6 +74,15 @@ func (p *proxy) handler(srv interface{}, stream grpc.ServerStream) error {
 	// Pass subject header upstream.
 	upstreamCtx = metadata.AppendToOutgoingContext(upstreamCtx, policy.ForwardedSubjectMD, sub)
 
+	// Dial upstream if necessary.
+	if p.upstreamConn == nil {
```

Comment:
Hm so if I see this correctly, after the initial dialing `p.upstreamConn` is never set to `nil`. So the `p.dialer` will only ever be invoked once and so the upstream certificates will never be refreshed?

Maybe this should additionally check if the connection is in a bad state and re-dial in that case? Of course the optimal thing would be to instead have a dynamic `RootCAs` but I don't think this is possible?

Reply:
Ah, good idea, I'll add a check.  Yes, as far as I've checked, it's not possible to dynamically change the certs once the connection has been established.